### ISSUE TITLE
Next steps from WW, BBL, NM meeting 9/18

### ISSUE THREAD

**Issue by stephpenn1** [PNNL-PREMIS/PREMIS-sapflow#37]
- [ ] **PACF:** Check is theres an error in Js calculation or numbers
- [ ] **Q1:** Calculate on ecosystem scale 
- [ ] **Q2:** Calculate on ecosystem scale
- [ ] **Q3:** Use better colors for sunny/cloudy; improve graphs
- [ ] **Overall analysis:** Calculate a `Js_peak` daily variable using Js from 11-1p

**Comment by bpbond** [comment]
>using Js from 11-1p

Or, Js for peak daylight hours, which might be more defensible (and not much harder).

**Comment by bpbond** [comment]
PACF
* Check calculation - are we doing for each tree separately? 
* Are data ordered correctly?
* Does this negative pacf show up for a single tree in a single week/month?

**Comment by bpbond** [comment]
Q1
* For each hour take average of Js and Rs across all trees

**Comment by bpbond** [comment]
Add sapwood area into Js
* Read Wenzhi area data, merge, multiply

**Comment by bpbond** [comment]
Calculate a Js_peak daily variable using Js from 11-1p
* Could do that (fixed hour) but weak
* Could do hours of maximum sunlight
* Could do _n_ largest Js values
* Could use `quantile()`

**Comment by stephpenn1** [comment]
@bpbond could you help with investigating the PACF errors and `Js_peak` calculation?

**Comment by stephpenn1** [comment]
How about we take use the top 3 hours of maximum sunlight via PAR to compute `Js_peak`? Should we do for each day, week, or month?

**Comment by bpbond** [comment]
@stephpenn1 I took a close look at what's going on with the PACF. Start by isolating just the first day of data:
```R
combined_pacf[38:60,] -> tst
plot(tst$Js_avg)
```
![Screen Shot 2020-10-08 at 6 01 30 AM](https://user-images.githubusercontent.com/1956468/95444380-bb90ba00-092b-11eb-9169-1f230b7e427f.png)

OK, this is nice and simple for us to examine now.
```R
pacf(tst$Js_avg, na.action = na.pass)
```
![Screen Shot 2020-10-08 at 6 02 44 AM](https://user-images.githubusercontent.com/1956468/95444564-f09d0c80-092b-11eb-87b0-520d27f13029.png)

This is reassuring--we're seeing a similar pattern to the mysterious (to us) pattern for the full dataset currently produced in the Rmd:
![Screen Shot 2020-10-08 at 6 05 43 AM](https://user-images.githubusercontent.com/1956468/95444875-538ea380-092c-11eb-9cdd-80b76aec6ff6.png)

The ACF gives us straight lagged correlations:
```
acf(tst$Js_avg)
```
![Screen Shot 2020-10-08 at 6 10 38 AM](https://user-images.githubusercontent.com/1956468/95445423-0232e400-092d-11eb-84e3-cb10a1b343d3.png)

**This makes sense.** Js is highly correlated with itself at a 1-hour lag, less correlated at 2, etc., and anti-correlated at 10-12 hours.

**Comment by bpbond** [comment]
So, we think the PACF plot is OK actually. How do we interpret it? From [Wikipedia](https://en.wikipedia.org/wiki/Partial_autocorrelation_function):

>In time series analysis, the partial autocorrelation function (PACF) gives the partial correlation of a stationary time series with its own lagged values, regressed the values of the time series at all shorter lags. It contrasts with the autocorrelation function, which does not control for other lags.

So in the PACF plot being produced by the RMarkdown file
* PACF at lag 1 is 0.94, meaning that Js values at a one-hour lag are highly correlated with each other. This makes sense.
* PACF at lag 2 is -0.68, meaning that Js values at a two-hour are anti-correlated _after the effects of the one-hour lag correlation are removed_. My interpretation: at a two-hour lag, we're likely to cross the midpoint (average Js) as the flux goes up in the morning or down in the afternoon, and so this produces a negative PACF at 2.

So in summary: I think we're OK here. The negative PACF at lag 2 is a mathematical consequence of the sharp diurnal cycle of Js.
